subprojects/docs/src/docs/userguide/kotlin_dsl.adoc

```diff
@@ -16,271 +16,124 @@
 
 [TIP]
 ====
-This user manual and link:{guidesUrl}[guides] contain build script excerpts that demonstrate both the Groovy DSL and the Kotlin DSL.
-This is the best place where to find how to do this and what with the Gradle Kotlin DSL ; and it covers all Gradle features from <<plugins#plugins, using plugins>> to <<customizing_dependency_resolution_behavior#customizing_dependency_resolution_behavior, customizing the dependency resolution behavior>>.
-
-If you are looking into migrating an existing build to the Gradle Kotlin DSL, please also check out the link:{guidesUrl}/migrating-build-logic-from-groovy-to-kotlin/[migration guide].
+If you are interested in migrating an existing Gradle build to the Kotlin DSL, please also check out the dedicated link:{guidesUrl}/migrating-build-logic-from-groovy-to-kotlin/[migration guide].
 ====
 
 
 [[sec:prerequisites]]
 == Prerequisites
 
 * The embedded Kotlin compiler is known to work on Linux, macOS, Windows, Cygwin, FreeBSD and Solaris on x86-64 architectures.
-* Knowledge of Kotlin syntax and basic language features is very helpful. The link:{kotlin-reference}[Kotlin reference documentation] and link:https://kotlinlang.org/docs/tutorials/koans.html[Kotlin Koans] should be useful to you.
-* Use of the <<plugins#sec:plugins_block, plugins {}>> block to declare Gradle plugins significantly improves the editing experience, and is highly recommended.
-
-
-[[sec:limitations]]
-== Limitations
-
-* There are some situations where the Kotlin DSL is slower. First use, on clean checkouts or ephemeral CI agents for example, link:{kotlin-dsl-issues}902[are known to be slower].
-The same applies to the scenario in which something in the _buildSrc_ directory changes, which invalidates build-script caching.
-The main bottleneck is script compilation.
-* In IntelliJ IDEA, you must link:https://www.jetbrains.com/help/idea/gradle.html#gradle_import[import your project from the Gradle model] to get content-assist and refactoring tools for Kotlin DSL scripts.
-* The Kotlin DSL will not support `model {}` elements. This is part of the link:https://blog.gradle.org/state-and-future-of-the-gradle-software-model[discontinued Gradle Software Model].
-Applying model rules from scripts is supported though, see the link:{kotlin-dsl-samples}model-rules[model-rules] sample.
-* Enabling the incubating <<multi_project_builds#sec:configuration_on_demand,configuration on demand>> feature is not recommended as it can lead to very hard-to-diagnose problems.
-* Functions of the Gradle API taking `Any` that accept lazy constructs such as `CopySpec.from(Any)` don't support Kotlin lambdas, use a `Callable { .. }` or a `provider { .. }` instead, see link:{kotlin-dsl-issues}1077[gradle/kotlin-dsl#1077].
-
-If you run to trouble or a suspected bug, please take advantage of the `gradle/kotlin-dsl` link:{kotlin-dsl-issues}[issue tracker].
-
-
-[[sec:ide_support]]
-== IDE support
```

Comment:
:x: I'm not comfortable with moving the "IDE support" section at the bottom of the document below Groovy interoperability and other advanced use cases, IDE experience is important

Reply:
I prefer to have it further down because it's not relevant to how one uses the Kotlin DSL. It's a "here's some additional info about the plugin" section. I thought this was reasonable because the section can easily be seen in the table of contents for the chapter.

I'm fine with moving it if you still feel strongly about it. For what it's worth, I would put the IDE support earlier if it were a magazine article or some other form of marketing.

Reply:
I'll just move the section, that's fine. On reflection, it probably is worth putting near the top as it's similar in some ways to prerequisites.